src/sources/aws_kinesis_firehose/mod.rs

```diff
@@ -820,4 +844,54 @@ mod tests {
         let response: models::FirehoseResponse = res.json().await.unwrap();
         assert_eq!(response.request_id, REQUEST_ID);
     }
+
+    #[tokio::test]
+    async fn event_service_token_passthrough_enabled() {
+        let (rx, address) = source(
+            None,
+            Some(vec!["an access key".to_string().into()]),
+            true,
+            Default::default(),
+            true,
+            true,
+        )
+        .await;
+
+        let timestamp: DateTime<Utc> = Utc::now();
+
+        spawn_send(
+            address,
+            timestamp,
+            vec![RECORD.as_bytes()],
+            Some("an access key"),
+            false,
+            Compression::None,
+        )
+        .await;
+
+        let events = collect_ready(rx).await;
+        let access_key = events[0].metadata().secrets().get("access_key").unwrap();
+        assert_eq!(access_key.to_string(), "an access key".to_string());
+    }
+
+    #[tokio::test]
+    async fn no_authorization_token_passthrough_enabled() {
```

Comment:
🤔 I wonder if it's worth a test for "no `access_keys` configured, but one was sent" - 
the converse shouldn't matter because the event should be rejected, I think.

Reply:
Actually the purpose of this test was to check that `store_access_keys` flag works properly: saves when enabled and has None when disabled. Should I rename it to make it more intuitive?

Reply:
Ah, I think these tests are fine (more or less after the token -> access_key rename, copied from splunk?) - more thinking if there's another combination worth testing. I think it's fine as-is and on-par with the comparable sources.